subsys/bluetooth/host/l2cap.c

```diff
@@ -839,10 +840,31 @@ int bt_l2cap_server_register(struct bt_l2cap_server *server)
 	return 0;
 }
 
+#ifdef CONFIG_BT_L2CAP_SEG_RECV
+static void l2cap_chan_seg_recv_rx_init(struct bt_l2cap_le_chan *chan)
+{
+	if (chan->rx.mps > BT_L2CAP_RX_MTU) {
+		LOG_ERR("Limiting RX MPS by stack buffer size.");
+		chan->rx.mps = BT_L2CAP_RX_MTU;
+	}
+
+	chan->_sdu_len = 0;
+	chan->_sdu_len_done = 0;
+}
+#endif /* CONFIG_BT_L2CAP_SEG_RECV */
+
 static void l2cap_chan_rx_init(struct bt_l2cap_le_chan *chan)
 {
 	LOG_DBG("chan %p", chan);
 
+	/* Redirect to experimental API. */
+	IF_ENABLED(CONFIG_BT_L2CAP_SEG_RECV, ({
```

Comment:
Does this have the same advantages as using `if (IS_ENABLED(CONFIG_BT_L2CAP_SEG_RECV))`?